### DISCUSSION TITLE
Vehicle models

### DISCUSSION THREAD

**Discussion by Yiquan-lol** [praveen-palanisamy/macad-gym#44] [Q&A]
Hi, 
I see that the vehicle is selected at random. Can I set the vehicle type？

**Comment by praveen-palanisamy** [comment]
Hi @Yiquan-lol ,
You can choose among: 4-wheeled vehicles (`vehicle_4w`), 2-wheeled vehicles (`vehicle_2w`) (and pedestrian) types for the vehicle actors using the config. For example:
https://github.com/praveen-palanisamy/macad-gym/blob/38884ac4bc7fb2e91a865950cff4eeadeed02f60/src/macad_gym/envs/hete/ncom/inde/po/intrx/ma/traffic_light_signal_1b2c1p_town03.py#L30-L32

If you want to change the model/make of a particular vehicle type, you can modify this line to pick the particular model/make you want from the available UE4 blueprint vehicle models in CARLA:

https://github.com/praveen-palanisamy/macad-gym/blob/38884ac4bc7fb2e91a865950cff4eeadeed02f60/src/macad_gym/carla/multi_env.py#L662

**Comment by praveen-palanisamy** [comment]
Converting this to a discussion for further discussion!
